src/OrchardCore.Modules/OrchardCore.ContentFields/Settings/DateTimeFieldSettingsDriver.cs

```diff
@@ -9,11 +11,18 @@ namespace OrchardCore.ContentFields.Settings
 {
     public class DateTimeFieldSettingsDriver : ContentPartFieldDefinitionDisplayDriver<DateTimeField>
     {
+        private readonly JsonSerializerOptions _jsonSerializerOptions;
+
+        public DateTimeFieldSettingsDriver(IOptions<JsonSerializerOptions> jsonSerializerOptions)
+        {
+            _jsonSerializerOptions = jsonSerializerOptions.Value;
+        }
+
         public override IDisplayResult Edit(ContentPartFieldDefinition partFieldDefinition)
         {
             return Initialize<DateTimeFieldSettings>("DateTimeFieldSettings_Edit", model =>
             {
-                var settings = partFieldDefinition.Settings.ToObject<DateTimeFieldSettings>();
+                var settings = partFieldDefinition.Settings.ToObject<DateTimeFieldSettings>(_jsonSerializerOptions);
```

Comment:
I don't think we need anything for types that are well-defined and not extensible. I'd even look at ways to use source code generation for these types to make things faster (STJ supports it).

`ContentItem`, `RecipeStep` are known to be able to contain polymorphic elements that are unknown at compile time so these ones need the custom options.

And even if a "setting" had to use a polymorphic property, it could use the option on a case by case basis. This one doesn't need it.

I would not mind keeping it if it was not making the code more complex, but here it means resolving a custom options object for nothing. And we don't even need special default options either like camelcase naming or indentation ...

Reply:
agreed. Removed what isn't needed